src/main/java/org/spongepowered/api/command/parameter/managed/standard/VariableValueParameters.java

```diff
@@ -0,0 +1,470 @@
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class VariableValueParameters {
+
+    private VariableValueParameters() {}
+
+    /**
+     * Creates a builder that can build a {@link ValueParameter} that returns
+     * an appropriate {@link CatalogType} from an argument.
+     *
+     * @return The builder
+     */
+    public static <T extends CatalogType> CatalogedTypeBuilder<T> catalogedElementParameterBuilder(Class<T> returnType) {
+        return Sponge.getRegistry().requireFactory(Factory.class).createCatalogedTypesBuilder(returnType);
+    }
+
+    /**
+     * Creates a builder that can build a {@link ValueParameter} that tries to
+     * match an argument against a fixed list of choices.
+     *
+     * <p>If the list of choices changes during the lifetime of the server,
+     * use {@link #dynamicChoicesBuilder(Class)} instead.</p>
+     *
+     * @param returnType The type of object that the resulting
+     *                   {@link ValueParameter} will return.
+     * @param <T> The type that will be returned
+     * @return The builder
+     */
+    public static <T> StaticChoicesBuilder<T> staticChoicesBuilder(Class<T> returnType) {
+        return Sponge.getRegistry().requireFactory(Factory.class).createStaticChoicesBuilder(returnType);
+    }
+
+    /**
+     * Creates a builder that can build a {@link ValueParameter} that tries to
+     * match an argument against a dynamic list of choices.
+     *
+     * <p>If the list of choices does not change during the lifetime of the
+     * server, use {@link #staticChoicesBuilder(Class)} instead.</p>
+     *
+     * @param returnType The type of object that the resulting
+     *                   {@link ValueParameter} will return.
+     * @param <T> The type that will be returned
+     * @return The builder
+     */
+    public static <T> DynamicChoicesBuilder<T> dynamicChoicesBuilder(Class<T> returnType) {
+        return Sponge.getRegistry().requireFactory(Factory.class).createDynamicChoicesBuilder(returnType);
+    }
+
+    /**
+     * Creates a builder that builds a {@link ValueParameter} that tries to
+     * match an a series of arguments with a supplied literal.
+     *
+     * @param returnType The type of object that the resulting
+     *                   {@link ValueParameter} will return.
+     * @param <T> The type that will be returned
+     * @return The builder
+     */
+    public static <T> LiteralBuilder<T> literalBuilder(Class<T> returnType) {
+        return Sponge.getRegistry().requireFactory(Factory.class).createLiteralBuilder(returnType);
+    }
+
+    /**
+     * Creates a builder that builds a {@link ValueParameter} that tries to
+     * construct a {@link Text} from an argument.
+     */
+    public static TextBuilder textBuilder() {
+        return Sponge.getRegistry().createBuilder(TextBuilder.class);
+    }
+
+    /**
+     * Creates a builder that builds a {@link ValueParameter} that tries to
```

Comment:
Javadoc outdated, this method no longer returns a builder

Reply:
Thanks, corrected.